File: packages/common-types/src/video.ts
export interface Video {
  id: number;
  title: string;
  description: string;
  coverUrl: string;
  videoUrl: string;
  duration: number;
  viewCount: number;
  danmakuCount: number;
  likeCount: number;
  coinCount: number;
  favoriteCount: number;
  shareCount: number;
  categoryId: number;
  userId: number;
  userName: string;
  userAvatar: string;
  tags: string[];
  createdAt: string;
}

export interface VideoUploadRequest {
  title: string;
  description: string;
  categoryId: number;
  tags: string[];
  cover: File;
  video: File;
}

export interface Category {
  id: number;
  name: string;
  parentId: number;
  icon: string;
  order: number;
}

File: apps/video-service/src/main/java/com/yuliyuli/video/entity/Video.java
package com.yuliyuli.video.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("t_video")
public class Video {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String description;
    private String coverUrl;
    private String videoUrl;
    private Integer duration;
    private Long viewCount;
    private Long danmakuCount;
    private Long likeCount;
    private Long coinCount;
    private Long favoriteCount;
    private Long shareCount;
    private Long categoryId;
    private Long userId;
    private String userName;
    private String userAvatar;
    private Integer status;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    @TableLogic
    private Integer deleted;
}
